java/src/main/java/io/grpc/examples/wallet/ExampleLoadBalancerProvider.java

```diff
@@ -0,0 +1,103 @@
+package io.grpc.examples.wallet;
+
+import io.grpc.LoadBalancer;
+import io.grpc.LoadBalancer.Helper;
+import io.grpc.LoadBalancerProvider;
+import io.grpc.LoadBalancerRegistry;
+import io.grpc.NameResolver;
+import io.grpc.NameResolver.ConfigOrError;
+import io.grpc.Status;
+import io.grpc.util.ForwardingLoadBalancer;
+import java.util.Map;
+import java.util.logging.Logger;
+
+
+/**
+ * This {@link LoadBalancerProvider} provides an example {@link LoadBalancer} implementation that
+ * delegates to round_robin and simply prints to System.out when it needs to handle resolved
+ * addresses. The purpose of this implementation is to demonstrate how to configure a custom
+ * {@link LoadBalancer} in Traffic Director, as explained in:
+ * https://cloud.google.com/traffic-director/docs/proxyless-configure-advanced-traffic-management
+ */
+public class ExampleLoadBalancerProvider extends LoadBalancerProvider {
+
+  @Override
+  public NameResolver.ConfigOrError parseLoadBalancingPolicyConfig(
+      Map<String, ?> rawLoadBalancingPolicyConfig) {
+    // The configuration map should have a "message" entry with a message that is printed out every
+    // time this load balancer handles resolved addresses.
+    String message = (String) rawLoadBalancingPolicyConfig.get("message");
```

Comment:
This would throw with bad config instead of returning an error. Generally all the contents of this method should be wrapped in a try-catch for RuntimeException.

Reply:
Checking for RuntimeException now.